client/ctx.go

```diff
@@ -0,0 +1,37 @@
+package client
+
+import (
+	"context"
+
+	"google.golang.org/grpc/metadata"
+)
+
+const (
+	logLevelRPCMetaKey = "log_level"
+	clientRequestIDKey = "client_request_id"
+
+	debugLevel = "debug"
+	infoLevel  = "info"
+	warnLevel  = "warn"
+	errorLevel = "error"
+)
+
+func WithDebugLogLevel(ctx context.Context) context.Context {
+	return metadata.AppendToOutgoingContext(ctx, logLevelRPCMetaKey, debugLevel)
+}
+
+func WithInfoLogLevel(ctx context.Context) context.Context {
+	return metadata.AppendToOutgoingContext(ctx, logLevelRPCMetaKey, infoLevel)
+}
+
+func WithWarnLogLevel(ctx context.Context) context.Context {
+	return metadata.AppendToOutgoingContext(ctx, logLevelRPCMetaKey, warnLevel)
+}
+
+func WithErrorLogLevel(ctx context.Context) context.Context {
+	return metadata.AppendToOutgoingContext(ctx, logLevelRPCMetaKey, errorLevel)
+}
```

Comment:
Maybe a unified method like WithLogLevel(ctx, logLevel) will do

Reply:
A unified interface is little more flexible，`logLevel` can be any string, may cause undefined problems

Reply:
Undefined log level problem could be handled with parameter validation. Anyway, this implement could work.